String manipulations giving grief. As usual.

### Version/revision of the library used
Latest and greatest self compiled build

### Describe the bug
The Compiler is nagging about 
`IRutils.cpp:98:50: error: conversion from 'const __FlashStringHelper* const' to 'const StringSumHelper' is ambiguous
     return kDashStr + uint64ToString(-input, base);`
My fix ( I do not know if is the best solution)
`return String(kDashStr) + uint64ToString(-input, base);`

Also 
```
ir_Coolix.cpp:614:57: error: operands to ?: have different types 'const __FlashStringHelper* const' and 'StringSumHelper'
           ? kOffStr : uint64ToString(getSensorTemp()) + 'C', kSensorTempStr);
```
My fix (also dont know if this is the best solution)
 `? kOffStr : String(uint64ToString(getSensorTemp()) + 'C'), kSensorTempStr);`


#### Expected behaviour
To compile.


Keep up the excelent work
Mutsos Gracias

--- a/src/IRutils.cpp
+++ b/src/IRutils.cpp
@@ -95,7 +95,10 @@ String uint64ToString(uint64_t input, uint8_t base) {
 /// @returns A String representation of the integer.
 String int64ToString(int64_t input, uint8_t base) {
   if (input < 0) {
-    return kDashStr + uint64ToString(-input, base);
+    // Using String(kDashStr) to keep compatible with old arduino
+    // frameworks. Not needed with 3.0.2.
+    ///> @see https://github.com/crankyoldgit/IRremoteESP8266/issues/1639#issuecomment-944906016
+    return String(kDashStr) + uint64ToString(-input, base);
   }
   return uint64ToString(input, base);
 }

--- a/src/ir_Coolix.cpp
+++ b/src/ir_Coolix.cpp
@@ -611,7 +611,11 @@ String IRCoolixAC::toString(void) const {
   result += addBoolToString(getZoneFollow(), kZoneFollowStr);
   result += addLabeledString(
       (getSensorTemp() == kCoolixSensorTempIgnoreCode)
-          ? kOffStr : uint64ToString(getSensorTemp()) + 'C', kSensorTempStr);
+          // Encasing with String(blah) to keep compatible with old arduino
+          // frameworks. Not needed with 3.0.2.
+          ///> @see https://github.com/crankyoldgit/IRremoteESP8266/issues/1639#issuecomment-944906016
+          ? kOffStr : String(uint64ToString(getSensorTemp()) + 'C'),
+      kSensorTempStr);
   return result;
 }
 

